### PR TITLE
Testing framework usability improvements

### PR DESCRIPTION
**Pull request by yrodiere**
Following this conversation: https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/quarkus-github-app

1. Mark the mocked GH client as offline.
2. Forward errors from the event handler to the testing thread.
3. ~~Raise errors when some mocks are not configured (missing `when(mock.someMethod()).then...`).~~ => Removed, see conversation below

### CODE REVIEW

**Comment by yrodiere** [comment]
So, yeah... exceptions are now thrown when there are missing mock behaviors, and that's the case in `quarkus-bot`. We need to update the tests there.

The alternative is to let mocks return some default value (e.g. `null`, an empty list, ...), which is the default behavior in Mockito. But then I can no longer give users a heads-up about missing mock behaviors...

**Comment by yrodiere** [comment]
Converting to drat while I'm waiting for some feedback. I'm really not sure the last commit is such a good idea...

**Comment by gsmet** [comment]
I think people will understand by themselves what's going wrong. At least that's my understanding. Probably better than having to mock a ton of things you don't use? Or is it less bad than what it looks?

Another option would be to have simple warnings. But again, we need to be sure that it really adds value.

**Comment by yrodiere** [comment]
Yes, I think you're right. People will just have to guess that an `NPE` in tests involving mocks propably means there was something wrong in their mocking.

I removed the last commit. The remaining changes should be safe, and improve the situation anyway, since we now throw the exception from the event handler directly in the testing thread (instead of just logging a warning...).

**Comment by yrodiere** [comment]
Looks like tests are passing. Ready for review :)

**Comment by gsmet** [comment]
Thanks, I will merge this and have a look at the additional beans issue tomorrow.

File: deployment/src/main/java/io/quarkiverse/githubapp/deployment/GitHubAppProcessor.java [hunk 1]
```diff
@@ -169,7 +169,7 @@ void additionalBeans(BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
                 UtilsProducer.class,
                 DefaultErrorHandler.class)
                 .setUnremovable()
-                .setDefaultScope(DotNames.SINGLETON)
+                .setDefaultScope(DotNames.APPLICATION_SCOPED)
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply deployment/src/main/java/io/quarkiverse/githubapp/deployment/GitHubAppProcessor.java:172]: I did that when trying to solve a problem related to mocking the error handler, but the solution turned out to be that my mock had to extend `DefaultErrorHandler` (and not `ErrorHandler` as one might expect).

I think this specific change can be reverted, and everything will still work fine: this is the *default* scope, for classes that do not define a scope through annotations, and `DefaultErrorHandler` *does* define a scope through annotations.

**Comment by yrodiere** [comment]
Thanks!

**Pull request merged**